daemon/policy.go

```diff
@@ -302,6 +304,79 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (newRev uint
 	return newRev, nil
 }
 
+type policyAddQueueOptions struct {
+	rules   policyAPI.Rules
+	opts    *AddOptions
+	context *policyQueueContext
+	reFetch bool
+}
+
+type policyQueueContext struct {
```

Comment:
What you are doing here is somewhat similar to the work I am doing in addition of a generic `EventQueue` in the following PR: https://github.com/cilium/cilium/pull/6943/commits/6f43d96c4d2ebad49d0fa4ad09ea6dccb1474827 . Perhaps in a future iteration we can add an EventQueue for policy management?

Reply:
yes! makes a lot of sense. This need to be backported to 1.4, so ;-)